src/cmake-tools.ts

```diff
@@ -152,6 +152,21 @@ export class CMakeTools implements vscode.Disposable, api.CMakeToolsAPI {
   private readonly _codeModel = new Property<codemodel_api.CodeModelContent|null>(null);
   private _codeModelDriverSub: vscode.Disposable|null = null;
 
+  private readonly _communicationModeSub = this.workspaceContext.config.onChange('cmakeCommunicationMode', () => {
+    log.info(localize('communication.changed.restart.driver', "Restarting the CMake driver after a communication mode change."));
+    return this._reloadCMakeDriver();
+  });
+
+  private readonly _generatorSub = this.workspaceContext.config.onChange('generator', () => {
+    log.info(localize('generator.changed.restart.driver', "Restarting the CMake driver after a generator change."));
+    return this._reloadCMakeDriver();
+  });
+
+  private readonly _preferredGeneratorsSub = this.workspaceContext.config.onChange('preferredGenerators', () => {
+    log.info(localize('preferredGenerator.changed.restart.driver', "Restarting the CMake driver after a preferredGenerators change."));
+    return this._reloadCMakeDriver();
+  });
+
```

Comment:
I expected to see watchers for configureArgs/Settings.  I noticed there is something in driver.ts that should be doing it, but there are still reports of those config settings not initiating a reload of the driver.  Are you able to repro that at all?

Reply:
No, I saw the reports and I tried but I wasn't able to reproduce. They seemed to update fine for me. I tried windows and linux, fileApi and serverApi. Can you also try to reproduce configureArgs/Settings not getting updated between a settings change and a configure?

Reply:
It looks like there may be a bug with deleting the whole setting not taking effect, but I'm not going to block this PR on that.  We can open a bug for it.  The important scenario of being able to add settings to a fresh folder appears to be working, so I'll approve.